### ISSUE TITLE
Hostile against Windows

### ISSUE THREAD

**Issue by chatchoi** [muon-build/muon#76]
Why are you hostile against the Windows platform and the MSVC compiler? You dismissed all PRs related to them.

**Comment by eli-schwartz** [comment]
I don't think that's an accurate analysis. Muon is primarily a single developer and done part-time, which means that @annacrombie isn't always available to review PRs. And for something that the primary developer doesn't personally use, it's somewhat expected that the main driver for implementing those features -- in this case, Windows support -- is going to be users submitting PRs.

As far as I know, the main person working on Windows support was @vtorri, who at some point suddenly closed any PRs he was working on and left. That's not really the same as @annacrombie "dismissing all PRs related to them".

**Comment by chatchoi** [comment]
> I don't think that's an accurate analysis. Muon is primarily a single developer and done part-time, which means that @annacrombie isn't always available to review PRs. And for something that the primary developer doesn't personally use, it's somewhat expected that the main driver for implementing those features -- in this case, Windows support -- is going to be users submitting PRs.
> 
> As far as I know, the main person working on Windows support was @vtorri, who at some point suddenly closed any PRs he was working on and left. That's not really the same as @annacrombie "dismissing all PRs related to them".

Quoted him:

> I found you on the list of contributors to the muon project. It seems you have given up on muon, don't you?

> yes, it seems that lattis is just ignoring my last PRs while not ignoring others, even if I wrote the windows port of muon... I'll do a muon fork with my last work on Windows compilers (cl, intel cpx, windows resource and other fixes)

**Comment by chatchoi** [comment]
>  Muon is primarily a single developer and done part-time

Btw, I can't understand why Debian managed to package Muon but not Meson, given the current status of Muon. I have to install `pipx` to install Meson on Debian. It's crazy.

**Comment by eli-schwartz** [comment]
> Quoted him:
> 
> > I found you on the list of contributors to the muon project. It seems you have given up on muon, don't you?
> 
> > yes, it seems that lattis is just ignoring my last PRs while not ignoring others, even if I wrote the windows port of muon... I'll do a muon fork with my last work on Windows compilers (cl, intel cpx, windows resource and other fixes)

I'm not really privy to the inner thoughts of either of them. I can imagine many possible explanations for this, some of them good, some of them bad. I prefer not to speculate.

As far as I'm aware, the muon developer remains interested in Windows support though.

> Btw, I can't understand why Debian managed to package Muon but not Meson, given the current status of Muon. I have to install `pipx` to install Meson on Debian. It's crazy.

Meson is definitely available on Debian.

https://packages.debian.org/source/bookworm/meson -- version 1.0.1
https://packages.debian.org/source/trixie/meson -- version 1.5.1

So is muon:
https://packages.debian.org/source/bookworm/muon-meson -- version 0.1.0
https://packages.debian.org/source/trixie/muon-meson -- version 0.2.0



Debian only packages new versions of software once every couple of years (unless you use the unstable development version of Debian), so I could understand needing to use pipx to get a *recent* version of Meson, but you should certainly be able to get an old version from Debian itself...

**Comment by vtorri** [comment]
On Mon, Sep 16, 2024 at 4:53 PM Eli Schwartz ***@***.***>
wrote:

> I don't think that's an accurate analysis. Muon is primarily a single
> developer and done part-time, which means that @annacrombie
> <https://github.com/annacrombie> isn't always available to review PRs.
> And for something that the primary developer doesn't personally use, it's
> somewhat expected that the main driver for implementing those features --
> in this case, Windows support -- is going to be users submitting PRs.
>
> As far as I know, the main person working on Windows support was @vtorri
> <https://github.com/vtorri>, who at some point suddenly closed any PRs he
> was working on and left.
>
several weeks with this answer for a 2 line PR : "looks good" and lattis
doesn't push. I've spent several weeks working on the windows module, and
lattis said he  would work on it differently (so my work is just sent to
trash)

so i'm fed up with that behavior. I'll do my own fork with commits for
Windows.

Vincent

**Comment by Tachi107** [comment]
> Btw, I can't understand why Debian managed to package Muon but not Meson, given the current status of Muon. I have to install `pipx` to install Meson on Debian. It's crazy.

I packaged muon for Debian because it's great, and might be helpful in some bootstrapping issues we have. (it still can't help because lots of Debian-specific build tools only know how to use Meson and not muon, and because muon can't currently build glib, but it might be helpful in the future nonetheless)

And as was mentioned earlier, Meson is obviously packaged too. How do you think we'd build software using Meson otherwise? ;)

**Comment by Tachi107** [comment]
$ apt search meson
    acr/testing,unstable 2.1.4-1 all
      autoconf like tool
    
    elpa-meson-mode/testing,unstable 0.3-2 all
      Major mode for the Meson build system files
    
    gi-docgen/testing,unstable 2024.1+ds-1 all
      source code documentation tool using GObject-Introspection
    
    gobject-introspection/testing,unstable 1.80.1-4 amd64
      Generate interface introspection data for GObject libraries
    
    meson/testing,unstable,now 1.5.1-2 all [installed]
      high-productivity build system
    
    muon-meson/experimental,now 0.2.0.418.gcafeb37c+ds-2 amd64 [installed]
      Meson implementation in C
    
    python3-mesonpy/testing,unstable 0.16.0-1 all
      Meson PEP 517 Python build backend

**Comment by chatchoi** [comment]
The last time I checked there is only `python3-mesonpy` that contains `meson` in its name, and `muon-meson`.

**Comment by eli-schwartz** [comment]
That is quite odd since the Debian Changelog for the "meson" package has as its first few entries:
```
meson (0.20.0-1) unstable; urgency=medium

  * New upstream version
  * Added build-deps to Mono, Flex and Bison as
    required by new unit tests

 -- Jussi Pakkanen <jpakkane@gmail.com>  Sat, 13 Sep 2014 00:46:56 +0300

meson (0.19.0-1) unstable; urgency=medium

  * New upstream release.
  * Meson now supports Fortran compilation,
    so build-depend on gfortran to test that it's working.

 -- Jussi Pakkanen <jpakkane@gmail.com>  Mon, 11 Aug 2014 00:24:16 +0300

meson (0.18.0-1) unstable; urgency=medium

  * New upstream release.
  * Made compilation and test output more verbose.

 -- Jussi Pakkanen <jpakkane@gmail.com>  Mon, 14 Jul 2014 18:26:28 +0300

meson (0.17.0-1) unstable; urgency=medium

  * Initial release. Closes: #712605, #712610

 -- Jussi Pakkanen <jpakkane@gmail.com>  Wed, 04 Jun 2014 22:33:44 +0300
```

And also, python3-mesonpy does NOT support `muon-meson`, it only supports `meson`. So it should be impossible to have a "python3-mesonpy" package but not have a "meson" one.

**Comment by eli-schwartz** [comment]
https://lists.debian.org/debian-testing-changes/2015/06/msg00001.html

> Testing migration summary 2015-06-01 (Monday)
> 
> On 2015-06-01 (Monday), the following source package(s) entered testing:
> 
> ```
>   meson                     0.24.0-1 (version)                  0.23.0-2 (previous)
> ```

Would be interested to know at what point you think meson was removed from the archives and then re-added later.
